### ISSUE TITLE
`prefer_self_in_static_references` rule should not trigger in some case.

### ISSUE THREAD

**Issue by r-plus** [realm/SwiftLint#3804]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

`prefer_self_in_static_references` opt-in rule will warn this type of code.

```
self.isMember(of: MyViewController.self)
```

If I change to `Self` to suppress warn, this `isMember(of:)` method return true even if `self` is inherited class instance.

I'm thinking  `prefer_self_in_static_references` rule should not trigger if accessing to `.self` to getting AnyClass object.

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? 0.45.1
* Installation method used (Homebrew, CocoaPods, building from source, etc)? CocoaPods
* Paste your configuration file:

```yml
opt_in_rules:
  - prefer_self_in_static_references
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)? No
* Which Xcode version are you using (check `xcodebuild -version`)? 13.2.1

**Comment by jpsim** [comment]
FYI @SimplyDanny in case you want to fix another class of false positives.

**Comment by SimplyDanny** [comment]
I'll take a look.

**Comment by prassyy** [comment]
If I may add another scenario of <ClassName>.self usage,
```
// Framework 1
class BaseViewController: UIViewController {
    static func create() -> UIViewController {
        let viewControllerName = String(describing: BaseViewController.self)
        let bundle = Bundle(for: BaseViewController.self)
        return UIViewController(nibName: viewControllerName, bundle: bundle)
    }
}

// Framework 2
class ChildViewController: BaseViewController { }
```
In the above case, the name of the nib should be `BaseViewController` and the bundle to look for it should be `Framework 1`. But since the swiftlint fixes the code from `BaseViewController.self` to `Self.self`, now the name of the viewController would be inferred as the ChildViewController and the bundle would be inferred as `Framework 2`. I don't think neither of it is wrong usage, it depends on where the developer has placed the resource which is very subjective to the project itself. 

Would it make more sense to ignore warning usages of `<ClassName>.self` if it is possible at all?

**Comment by SimplyDanny** [comment]
> Would it make more sense to ignore warning usages of `<ClassName>.self` if it is possible at all?

I think, this makes sense. To make each usage of `<ClassName>.self` explicit sounds reasonable. Thank you for bringing this case up!
